dependencies/default/requirements.txt

```diff
@@ -1,3 +1,3 @@
 # Always adjust install_requires in setup.cfg and pytest-min-requirements.txt
 # when changing runtime dependencies
-pytest >= 7.0.0,<8
+pytest >= 7.0.0, !=8.0.0rc1, !=8.0.0rc2
```

Comment:
I ran your patch successfully against pytest-8.0.0rc1 and pytest-8.0.0rc2. This means there's not reason to exclude them in the compatible versions. It's more likely that #764 reduced the number of issues with the collection changes in pytest.

More importantly, we should try to keep the upper version bound at all costs, both in `requirements.txt` and especially in `setup.py`. Otherwise we assume that pytest-asyncio is compatible with potentially breaking changes in pytest 9.